gradio/route_utils.py

```diff
@@ -51,6 +52,9 @@
     from gradio.routes import App
 
 
+config_lock = threading.Lock()
```

Comment:
shouldn't this be an asyncio.Lock?

Reply:
yeah I think a threading Lock won't work because jobs are run all on the same thread (same event loop), so this could still run the code simultaneously across different async tasks.

Reply:
jk this is not run from the queue, thread lock is correct